### PR TITLE
A function SHOULD NOT return ctx.Err() if the method was not interrupted because of the context passed to a function 

### PR DESCRIPTION
**Pull request by Kaushal28**
A function SHOULD NOT return ctx.Err() if the method was not interrupted because of the context passed to a function

A function should return ctx.Err() if the context is cancelled, except if there is no work in progress. In that case, it should return nil.

ctx.Err() is returned usually in function that has the "processing logic"

### CODE REVIEW

File: exporters/jaeger/jaeger.go [hunk 1]
```diff
@@ -115,11 +115,6 @@ func (e *Exporter) ExportSpans(ctx context.Context, spans []sdktrace.ReadOnlySpa
 func (e *Exporter) Shutdown(ctx context.Context) error {
 	// Stop any active and subsequent exports.
 	e.stopOnce.Do(func() { close(e.stopCh) })
-	select {
```
Comment (Kaushal28) [inline exporters/jaeger/jaeger.go:118]: Removed this as downstream `shutdown` function handles context and this function does not have processing part.

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go [hunk 2]
```diff
@@ -149,7 +149,7 @@ func (c *client) Shutdown(ctx context.Context) error {
 
 	c.requestFunc = nil
 	c.httpClient = nil
-	return ctx.Err()
```
Comment (Kaushal28) [inline exporters/otlp/otlpmetric/otlpmetrichttp/client.go:152]: There's no "processing" being done here. But not sure if this is correct.

**Comment by Kaushal28** [comment]
@pellared , I have done 1 change and fixed corresponding test case. Please have a look and let me know if this makes sense.

File: exporters/jaeger/uploader.go [hunk 3]
```diff
@@ -297,7 +297,7 @@ var _ batchUploader = (*collectorUploader)(nil)
 func (c *collectorUploader) shutdown(ctx context.Context) error {
 	// The Exporter will cancel any active exports and will prevent all
 	// subsequent exports, so nothing to do here.
-	return nil
+	return ctx.Err()
```
Comment (pellared) [inline exporters/jaeger/uploader.go:300]: This should return an error with an active export was interrupted because of the cancelation.

**Comment by MrAlias** [comment]
This PR has been in a draft state for many months without any contribution or response from the author. Closing.

Please finalize your ideas in your branch prior to re-proposing these changes.

**Pull request closed without merging**